allennlp/tests/semparse/contexts/table_question_context_test.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=no-self-use,invalid-name,protected-access,too-many-public-methods
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.semparse.contexts import TableQuestionContext
+from allennlp.data.tokenizers import WordTokenizer
+from allennlp.data.tokenizers.word_splitter import SpacyWordSplitter
+
+MAX_TOKENS_FOR_NUMBER = 6
+
+class TestTableQuestionContext(AllenNlpTestCase):
+    def setUp(self):
+        super().setUp()
+        self.tokenizer = WordTokenizer(SpacyWordSplitter(pos_tags=True))
+
+    def test_number_extraction(self):
+        question = """how many players on the 191617 illinois fighting illini men's basketball team
+                      had more than 100 points scored?"""
+        question_tokens = self.tokenizer.tokenize(question)
+        test_file = f'{self.FIXTURES_ROOT}/data/corenlp_processed_tables/TEST-7.table'
+        table_question_context = TableQuestionContext.read_from_file(test_file, question_tokens)
+        _, number_entities = table_question_context.get_entities_from_question()
+        assert number_entities == [("191617", "191617"), ("100", "100")]
```

Comment:
Seems more useful to have a token index, not a string, for what triggered the number.  There might be more than one matching string.  Also, I know we talked at one point about separating out date numbers from non-date numbers.  Not sure if that's a good idea or not...  I suppose we can worry about it later, in any event.

Reply:
Changed token strings to indices.